lib/compat.php

```diff
@@ -24,3 +24,39 @@ function gutenberg_safe_style_css_column_flex_basis( $attr ) {
 	return $attr;
 }
 add_filter( 'safe_style_css', 'gutenberg_safe_style_css_column_flex_basis' );
+
+/**
+ * Adds an inline script to augment the default persistence method for data to
+ * use WordPress client user settings.
+ *
+ * @since 5.8.0
+ */
+function gutenberg_settings_cookie_persistence() {
+	// Ensure `utils` is a dependency, which makes available `userSettings`,
+	// `getUserSetting`, and `setUserSettings` window globals.
+	$data_script = wp_scripts()->query( 'wp-data', 'registered' );
+	if ( ! in_array( 'utils', $data_script->deps ) ) {
+		$data_script->deps[] = 'utils';
+	}
+
+	wp_add_inline_script(
+		'wp-data',
+		"
+( function() {
+	wp.data.use( wp.data.plugins.persistence, {
+		storage: {
+			getItem: function() {
+				return (
+					window.getUserSetting( 'dataState' ) ||
+					localStorage.getItem( 'WP_DATA_USER_' + window.userSettings.uid )
+				);
+			},
+			setItem: function( key, value ) {
+				window.setUserSetting( 'dataState', value );
```

Comment:
Related implementation of the method:

https://github.com/WordPress/wordpress-develop/blob/8347d7e22f98520228309392757ea75fd7bca4db/src/js/_enqueues/lib/cookies.js